### PR TITLE
Remove Arrow-kt dependency

### PR DESCRIPTION
**Pull request by bmarty**
Remove Arrow-kt dependency.

 
## Type of change

- [ ] Feature
- [ ] Bugfix
- [ ] Technical
- [x] Other : dep removal.

## Tests

- Smoke test (in progress)

## Tested devices

- [ ] Physical
- [ ] Emulator
- OS version(s):

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [ ] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

**Comment by ganfra** [comment]
I think we should get rid of Arrow as it's just used for some Option. We have our own simple Optional type in the sdk, could we replace?

**Comment by bmarty** [comment]
OK, I have removed this dep. Thanks to review again the PR!

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/util/Optional.kt [hunk 1]
```diff
@@ -33,23 +31,19 @@ data class Optional<T : Any> constructor(private val value: T?) {
         }
     }
 
-    fun getOrElse(fn: () -> T): T {
+    fun orElse(fn: () -> T): T {
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/util/Optional.kt:34]: (not used at the moment)

File: matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/util/Optional.kt [hunk 2]
```diff
@@ -33,23 +31,19 @@ data class Optional<T : Any> constructor(private val value: T?) {
...
+        fun <T : Any> from(value: T?): Optional<T> = Optional(value)
 
-        fun <T : Any> empty(): Optional<T> {
-            return Optional(null)
-        }
+        fun <T : Any> empty(): Optional<T> = Optional(null)
     }
 }
 
+fun <T : Any> T?.toOption() = Optional(this)
```
Comment (bmarty) [inline matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/util/Optional.kt:47]: (another new alias)

**Review by ganfra — APPROVED** [review]
Thanks for this, LGTM

**Pull request merged**